lib/configuration.js

```diff
@@ -15,150 +15,614 @@
+   * @memberof Configuration
+   * @private
+   * @type {Boolean}
+   * @defaultvalue true
+   */
+  this._reportUncaughtExceptions = true;
+  /**
+   * The _shouldReportErrorsToAPI property is meant to denote whether or not
+   * the Stackdriver error reporting library will actually try to report Errors
+   * to the Stackdriver Error API. The value of this property is derived from
+   * the `NODE_ENV` environmental variable. If the `NODE_ENV` variable is set to
+   * 'production' then the _shouldReportErrorToAPI property will be set to true
+   * and error reporting library will attempt to send errors to the Error API.
+   * Otherwise the value will remain false and errors will not be reported to
+   * the API.
+   * @memberof Configuration
+   * @private
+   * @type {Boolean}
+   */
+  this._shouldReportErrorsToAPI = env.NODE_ENV === 'production';
+  /**
+   * The _projectId property is meant to contain the string project id that the
+   * hosting application is running under. The project id is a unique string
+   * identifier for the project. If the Configuration instance is not able to
+   * retrieve a project id from the metadata service or the runtime-given
+   * configuration then the property will remain null. If given both a project
+   * id through the metadata service and the runtime configuration then the
+   * instance will assign the value given by the metadata service over the
+   * runtime configuration. If the instance is unable to retrieve a valid
+   * project id or number from runtime configuration and the metadata service
+   * then this will trigger the `error` event in which listening components must
+   * operate in 'offline' mode.
+   * {@link https://cloud.google.com/compute/docs/storing-retrieving-metadata}
+   * @memberof Configuration
+   * @private
+   * @type {String|Null}
+   * @defaultvalue null
+   */
+  this._projectId = null;
+  /**
+   * The _projectNumber property is meant to contain the number project id that
+   * the hosting application is running under. The project number is a unique
+   * number identifier for the project. If the Configuration instance is unable
+   * to retrieve a project number from the metadata service or the runtime
+   * configuration then the property will remain null. If given both a project
+   * number through the metadata service and the runtime configuration then the
+   * instance will assign the value given by the metadata service over the
+   * runtime configuration. If the Configuration instance is not able to 
+   * retrieve a project number or a project id from runtime configuration or
+   * from the metadata service then this will trigger the `error` event in which
+   * listening components must operate in 'offline' mode.
+   * {@link https://cloud.google.com/compute/docs/storing-retrieving-metadata}
+   * @memberof Configuration
+   * @private
+   * @type {String|Null}
+   * @defaultvalue null
+   */
+  this._projectNumber = null;
```

Comment:
@ofrobots This field actually stores the project number since that is what is released on the diagnostics api right now. When we do a release with the projectId get method then the projectId field will be set